hs_collection_resource/views.py

```diff
@@ -0,0 +1,105 @@
+import logging
+
+from django.http import JsonResponse
+from django.db import transaction
+
+from hs_core.views.utils import authorize, ACTION_TO_AUTHORIZE
+from hs_core.hydroshare.utils import get_resource_by_shortkey, resource_modified
+
+logger = logging.getLogger(__name__)
+
+
+# update collection
+def update_collection(request, shortkey, *args, **kwargs):
+    """
+    Add resources to a collection. The POST request should contain a
+    list of resource ids for those resources to be part of the collection. Any existing resources
+    from the collection are removed before adding resources as specified by the list of
+    resource ids in the post request. Requesting user must at least have metadata view permission
+    for any new resources being added to the collection.
+
+    :param shortkey: id of the collection resource to which resources are to be added.
+    """
+
+    status = "success"
+    msg = ""
+    metadata_status = "Insufficient to make public"
+    try:
+        with transaction.atomic():
+            collection_res_obj, is_authorized, user = authorize(request, shortkey,
+                                                                needed_permission=ACTION_TO_AUTHORIZE.EDIT_RESOURCE)
+
+            if collection_res_obj.resource_type != "CollectionResource":
+                raise Exception("Resource {0} is not a collection resource.".format(shortkey))
+
+            # get res_id list from POST
+            updated_contained_res_id_list = request.POST.getlist("resource_id_list")
+
+            if len(updated_contained_res_id_list) > len(set(updated_contained_res_id_list)):
+                raise Exception("Duplicate resources were found for adding to the collection")
+
+            # check authorization for all new resources being added to the collection
+            # the requesting user should at least have metadata view permission for each of the
+            # new resources to be added to the collection
+            for updated_contained_res_id in updated_contained_res_id_list:
+                if not collection_res_obj.resources.filter(short_id=updated_contained_res_id).exists():
+                    res_to_add, _, _ = authorize(request, updated_contained_res_id,
+                                                 needed_permission=ACTION_TO_AUTHORIZE.VIEW_METADATA)
+
+                    # for now we are not allowing a collection resource to be added to another collection resource
+                    if res_to_add.resource_type == "CollectionResource":
+                        raise Exception("Resource {0} is a collection resource which can't be added.".format(shortkey))
```

Comment:
@zhiyuli This confirms my statement above on your testing code that nesting collections are not allowed. So those corresponding test cases should be removed but rather we only test that nesting collections are not allowed, i.e., an exception will be raised if doing so.